gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/vertx/web/handler/impl/CookieSessionHandler.java

```diff
@@ -121,10 +123,23 @@ public void handle(RoutingContext context) {
         }
 
         // Need to wait the session to be ready before invoking next.
-        sessionObs
-                .doOnError(t -> logger.warn("Unable to restore the session", t))
-                .doFinally(context::next)
-                .subscribe();
+        try {
+            sessionObs
+                    .doFinally(context::next)
+                    .subscribe(
+                            success -> logger.trace("Session restored successfully"),
+                            error -> {
+                                final Throwable cause = error.getCause();
+                                if (cause instanceof PrematureJWTException | error instanceof ExpiredJWTException) {
+                                    logger.info("Unable to restore the session: {}", cause.getMessage());
+                                } else {
+                                    logger.warn("Unable to restore the session: {}", cause.getMessage());
+                                }
+                            }
+                    );
+        } catch (Exception ex) {
+            logger.error("Some exception reported");
```

Comment:
**todo:** I don't see where an exception can be thrown here, can you explain ? If an exception can be raised, I think the message should be changed and the exception should be provided as parameter of the error method to log the stacktrace as it will be an unexpected error. wdyt ?

Reply:
This try catch is pointless. Removed.